### ISSUE TITLE
Occasionally more transfers than granted are delivered.

### ISSUE THREAD

**Issue by kjnilsson** [rabbitmq/rabbitmq-server#2597]
Whilst testing the AMQP 1.0 client I discovered that the AMQP 1.0 plugin has tendency to sometimes deliver more transfers than granted through the flow control mechanism resulting in strict clients closing the link with a `transfer-limit-exceeded` error. This is likely to be due to how the `x-credit` parameter works in rabbit itself.

**Comment by woppa684** [comment]
Was this problem ever solved? We might be experiencing the same problem at the moment.

**Comment by woppa684** [comment]
> The issue is still open, thus it is not solved.

Of course that could be an error, but let's not assume that :) Any idea where I need to start looking in the code to find the problem? I know the 1.0 implementation is not perfect at the moment, but it's usable for us, except for a few small problems like this one. So maybe we can contribute fixes.

**Comment by lukebakken** [comment]
@kjnilsson ^^^^ thanks

**Comment by ansd** [comment]
Yes, this issue is fixed, see point 32 of https://github.com/rabbitmq/rabbitmq-server/pull/9022#issue-1839124726
